th/lessons/basics/pattern-matching.md

```diff
@@ -0,0 +1,100 @@
+---
+version: 1.0.1
+title: การจับคู่รูปแบบ (Pattern Matching)
+redirect_from:
+  - /lessons/basics/pattern-matching/
+---
+
+Pattern Matching เป็นส่วนที่ทรงพลังมากส่วนนึงของ Elixir มันทำให้เราสามารถเปรียบเทียบค่า, โครงสร้างข้อมูล หรือแม้แต่ฟังก์ชันได้ ในบทนี้เราจะมาดูกันว่า Pattern Matching มันใช้งานอย่างไร
+
+{% include toc.html %}
+
+## Match Operator
+
+ใน Elixir เครื่องหมาย `=` จริง ๆ แล้วมันคือ เครื่องหมายการจับคู่ (Match Operator)  เปรียบได้กับเครื่องหมาย "เท่ากับ" ในพีชคณิต การใช้เครื่องหมาย `=` จะทำให้ Expression นั้น ๆ กลายเป็นสมการ และทำให้ Elixir ทำการจับคู่ค่าฝ่ังซ้ายกับค่าฝั่งขวาทันที ถ้าสำเร็จ มันก็จะคืนค่าของสมการนั้นออกมา แต่ถ้าไม่สำเร็จ มันก็จะคืนค่าเออเรอร์ออกมาแทน เราลองมาดูตัวอย่างกัน:
+
+```elixir
+iex> x = 1
+1
+```
+
+คราวนี้ลองทำการจับคู่ดู
+
+```elixir
+iex> 1 = x
+1
+iex> 2 = x
+** (MatchError) no match of right hand side value: 1
+```
+
+ลองจับคู่กับ Collections ที่เราเพิ่งเรียนมา
```

Comment:
Collections -> collection

I don't think we need to have plural in Thai translation since we don't have that concept in Thai so I think just make everything singular would be more natural when mixing with the rest.